camayoc/tests/qpc/ui/test_credentials.py

```diff
@@ -86,22 +135,111 @@ def test_create_delete_credential(browser, qpc_login, credential_type):
         4) Delete the newly created credential.
     :expectedresults: A new credential is created and then deleted.
     """
-    name = uuid4()
-    username = uuid4()
-    password = uuid4()
-    create_credential(browser, credential_type, name, username, password)
-    # Confirmation alert sometimes covers checkboxes and buttons when clicking
-    checkbox = Checkbox(browser, locator=Locator(
-        xpath=checkbox_xpath(name)))
-    try:
-        checkbox.fill(True)
-    except WidgetOperationFailed:
-        clear_toasts(view=browser)
-        checkbox.fill(True)
-    Button(browser, 'Delete').click()
-    DeleteModalView(browser, locator=Locator(
-                        css='.modal-content')).delete_button.click()
-    # Checkbox next to name of credential is used to check for existence
-    with pytest.raises(
-            (NoSuchElementException, StaleElementReferenceException)):
-        browser.element(locator=Locator(xpath=checkbox_xpath(name)))
+    name = utils.uuid4()
+    username = utils.uuid4()
+    password = utils.uuid4()
+    create_credential(browser, {
+        'name': name,
+        'username': username,
+        'password': password,
+        'credential_type': credential_type
+        })
+    set_checkbox(browser, name, True)
+    delete_credential(browser, name)
+
+
+def test_create_delete_credential_optional(browser, qpc_login):
+    """Create and then delete a credential with optional parameters.
+
+    :id: 37632616-86e9-47d1-b1f6-78dd5dde0774
+    :description: Optional parameters are included in this test,
+        like Become User and Become Password. Afterwards, the new
+        credential is deleted.
+    :steps:
+        1) Log into the UI.
+        2) Go to the credentials page and open the Add Credential modal.
+        3) Fill in required and optional fields and save.
+        4) Delete the newly created credential.
+    :expectedresults: A new credential is created and then deleted.
+    """
+    name = utils.uuid4()
+    username = utils.uuid4()
+    password = utils.uuid4()
+    become_user = utils.uuid4()
+    become_pass = utils.uuid4()
+    create_credential(browser, {
+        'name': name,
+        'username': username,
+        'password': password,
+        'credential_type': 'Network',
+        'become_user': become_user,
+        'become_pass': become_pass
+        })
+    set_checkbox(browser, name, True)
```

Comment:
Is this set_checkbox used to select the credentials do delete?

Reply:
yes. there isn't a select all button so I manually select the rows I want to delete (in this case based on their `name`, then  go through the deletion process.